lib/resty/core/regex.lua

```diff
@@ -154,6 +155,19 @@ local function get_max_regex_cache_size()
 end
 
 
+local regex_cache_is_empty = true
+
+function _M.is_regex_cache_empty()
+    return regex_cache_is_empty
+end
+
+
+local function lrucache_set_wrapper(...)
+    _M.regex_cache_is_empty = false
```

Comment:
Let's avoid this table lookup in `_M`. We should directly reference the upvalue.